### PR TITLE
Properly calculate RHS for pressure equation on GPU.

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Thermal bubble regression test now works again. Dumb leftover mistake from #228. Inb4 we should have unified the poisson solvers.

Love it when you're off by 10^50:

Before:
```
[ Info: Δu: min=-3.55764e+54, max=3.55764e+54, mean=-2.0648e+35, absmean=6.04154e+52, std=2.66887e+53
[ Info: Δv: min=-3.55764e+54, max=3.55764e+54, mean=2.57926e+35, absmean=6.04154e+52, std=2.66887e+53
[ Info: Δw: min=-8.03359e+54, max=5.86492e+54, mean=-8.09207e+35, absmean=7.70596e+52, std=3.93725e+53
[ Info: ΔT: min=-9.12137e+52, max=3.63655e+52, mean=1.50945e+34, absmean=3.54008e+50, std=3.58814e+51
[ Info: ΔS: min=-3.2411e+53, max=1.29218e+53, mean=2.84495e+34, absmean=1.2579e+51, std=1.27498e+52
```

After:
```
[ Info: Δu: min=-1.20346e-17, max=1.11131e-17, mean=-3.21903e-22, absmean=6.46707e-19, std=1.21175e-18
[ Info: Δv: min=-1.8052e-17, max=1.07099e-17, mean=-3.80102e-22, absmean=6.51696e-19, std=1.25463e-18
[ Info: Δw: min=-3.97428e-17, max=3.5345e-17, mean=4.25729e-21, absmean=3.64351e-18, std=5.66798e-18
[ Info: ΔT: min=-1.77636e-15, max=1.77636e-15, mean=-4.77049e-18, absmean=1.95156e-17, std=1.86152e-16
[ Info: ΔS: min=0, max=0, mean=0, absmean=0, std=0
```

### CODE REVIEW

File: src/time_steppers.jl [hunk 1]
```diff
@@ -220,9 +220,9 @@ function calculate_poisson_right_hand_side!(::GPU, grid::Grid, Δt, u, v, w, Gu,
                 # Calculate divergence of the RHS source terms (Gu, Gv, Gw) and applying a permutation
                 # which is the first step in the DCT.
                 if CUDAnative.ffs(k) == 1  # isodd(k)
-                    @inbounds RHS[i, j, convert(UInt32, CUDAnative.floor(k/2) + 1)] = div_f2c(grid, u, v, w, i, j, k) + Δt*div_f2c(grid, Gu, Gv, Gw, i, j, k)
+                    @inbounds RHS[i, j, convert(UInt32, CUDAnative.floor(k/2) + 1)] = div_f2c(grid, u, v, w, i, j, k) / Δt + div_f2c(grid, Gu, Gv, Gw, i, j, k)
```
Comment (glwagner) [inline src/time_steppers.jl]: 😱

File: test/test_dynamics.jl [hunk 2]
```diff
@@ -154,7 +154,6 @@ function internal_wave_test(; N=128, Nt=10)
      N = 128
     Δt = 0.01 * 1/σ
 
-    @show σ/f
```
Comment (glwagner) [inline test/test_dynamics.jl]: 👌

**Comment by ali-ramadhan** [comment]
> this doesn't explain why the test in #238 fails, but it is good that this test passes now. Does it makes sense to check that the RB test passes before merging this?

Yeah no this isn't related to #238.

That's a good idea, I'm also waiting to make sure that GitLab CI passes.

#239 is now merged so I'll rebase this branch on top of master and run the tests again to see if RB passes.

**Comment by glwagner** [comment]
It's extremely confusing to me how this test can pass but #238 does not. Presumably there is a problem with the test in #238.

**Comment by ali-ramadhan** [comment]
GitLab CI passes and so does the RB regression test so I'll merge this now.

**Pull request merged**